src/storage/kv/cursor.rs

```diff
@@ -147,7 +164,16 @@ impl<I: Iterator> Cursor<I> {
             );
         }
         if !self.valid()? {
-            self.max_key = Some(key.as_encoded().to_owned());
+            // Do not set `max_key` when prefix seek enabled.
+            // `max_key` is used to record the last key before the iter reaches
+            // the end, so later seek whose key is larger than `max_key` can be
+            // return not found directly instead of calling underlying iterator
+            // to seek the key.
+            // But when prefix seek enabled, !valid() doesn't mean reaching the
+            // end of iterator range. So the `max_key` shouldn't be updated.
+            if !self.prefix_seek {
```

Comment:
Should check it when setting `min_key`?

Reply:
nice catch! let me think about it and add a test to cover this